src/native/fs/fs_napi.cpp

```diff
@@ -213,6 +224,258 @@ struct Stat : public FSWorker
+        if (r) {
+            SetSyscallError();
+            return;
+        }
+        auto start_ctime = stat_res.st_ctime;
+
+        buf = static_cast<char*>(malloc(buf_len));
+        if (buf == NULL) {
+            SetSyscallError();
+            return;
+        }
+
+        #if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
+            buf_len = listxattr(_path.c_str(), buf, buf_len, 0);
+        #else
+            buf_len = listxattr(_path.c_str(), buf, buf_len);
+        #endif
+
+        if (buf_len == -1) {
+            SetSyscallError();
+            free(buf);
+            return;
+        }
+
+        key = buf;
+        while (buf_len > 0) {
+            // strncmp shouldn't cause error if length of key < USER_MD_PREFIX
+            if (strncmp(USER_MD_PREFIX.c_str(), key, USER_MD_PREFIX.length()) != 0)
+            {
+                key_len = strlen(key) + 1;
+                buf_len -= key_len;
+                key += key_len;
+                continue;
+            }
+
+            #if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
+                value_len = getxattr(_path.c_str(), key, NULL, 0, 0, 0);
+            #else
+                value_len = getxattr(_path.c_str(), key, NULL, 0);
+            #endif
+
+            if (value_len == -1) {
+                SetSyscallError();
+                break;
+            }
+
+            if (value_len > 0) {
+                val = static_cast<char*>(malloc(value_len + 1));
+                if (val == NULL) {
+                    break;
+                }
+
+                #if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
+                    value_len = getxattr(_path.c_str(), key, val, value_len, 0, 0);
+                #else
+                    value_len = getxattr(_path.c_str(), key, val, value_len);
+                #endif
+
+                if (value_len == -1) {
+                    SetSyscallError();
+                    break;
+                }
+                
+                val[value_len] = 0;
+                _md.push_back(KeyValue{ 
+                    std::string(key),
+                    std::string(val)
+                });
+
+                free(val);
+                val = NULL;
+            } else if (value_len == 0) {
+                _md.push_back(KeyValue{ 
+                    std::string(key),
+                    std::string()
+                });
+            }
+
+            key_len = strlen(key) + 1;
+            buf_len -= key_len;
+            key += key_len;
+        }
+
+        if (buf) free(buf);
+        if (val) free(val);
+
+        r = stat(_path.c_str(), &stat_res);
+        if (r) {
+            SetSyscallError();
+            return;
+        }
+        auto end_ctime = stat_res.st_ctime;
+        if (start_ctime != end_ctime) {
+            // Cancel on ctime change due to non atomic nature of this call
+            errno = ECANCELED;
+            SetSyscallError();
```

Comment:
This is not a syscall error so why overloading this ECANCELED error?
Why not just use a SetError?